core/domain/stats_domain.py

```diff
@@ -48,19 +59,23 @@ def get_multi_by_multi_explorations(
         """Gets domain objects given a list of exploration and state tuples.
 
         Args:
-            exploration_state_list: a list of exploration ID and state name
-                tuples
-            rule_str_list: a list of rule spec strings which are used to filter
-                the answers matched to the provided explorations and states
-
-        Returns a list of StateRuleAnswerLog objects containing answers matched
-        to each exploration ID-state name. The number of elements returned is
-        the same as the number of elements passed in exploration_state_list. The
-        number of answers in each StateRuleAnswerLog object depends on the
-        number of answers matched against each rule spec strings specified in
-        rule_str_list.
+            exploration_state_list: list(tuple). A list of exploration ID and
+                state name tuples.
+            rule_str_list: list. A list of rule spec strings which are used to
```

Comment:
Actually I was not able to figure out what exactly does rule spec string do. I traced it back to the point where it was being set but couldn't figure out what does it do. I tried printing it in various situations but the value was `Default` everytime which is why I couldn't see the various values it takes.

If you could just give me an idea or tell where to look it would be helpful.